Refactor/lint onboarding slides

daningenthron opened the pull request:
## What type of PR is this? (check all applicable)

- [ X ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Documentation Update

## Description

From #2470 - cleans up ESlint errors that weren't passing Husky. For the most part these were small edits (snake case to camel case etc), although two changes may need some attention:

- `EmailListTermsConditionsForm.jsx` contained a `dangerouslySetInnerHTML` warning that wouldn't pass; this has been bypassed by disabling the ESLint rule for this line. I think a better fix could be adding the DOMPurify dependency and sanitizing the HTML here and adding via `innerHTML` vs `dangerouslySetInnerHTML`, but those changes didn't seem in line with this PR.

- In `SlideContent.jsx` I set up a blank (non-validating) shape for the style PropType. Not sure if this is best practice, but it seemed to make sense as far as trying to appease ESLint without defining the CSS attributes (and their types) ahead of time.

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ X ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

![image](https://media.giphy.com/media/1nP7ThJFes5pgXKUNf/giphy.gif)

![alt_text](gif_link) Hand pulling lint out of a lint trap in a single super-thick sheet

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/thepracticaldev/dev.to?pullRequest=4240) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/thepracticaldev/dev.to?pullRequest=4240) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/thepracticaldev/dev.to?pullRequest=4240) it.</sub>

maestromac commented on app/javascript/onboarding/components/EmailListTermsConditionsForm.jsx:
```diff
@@ -13,10 +13,10 @@ class EmailTermsConditionsForm extends Component {
+      checkedCodeOfConduct: false,
+      checkedTermsAndConditions: false,
+      emailMembershipNewsletter: true,
+      emailDigestPeriodic: true,
```
Not that this will technically be breaking the onboard process, but it won't properly update user's preference. This is because the [whole state is sent via fetch](https://github.com/thepracticaldev/dev.to/blob/16e17843ed85e3095b12437fc2c9e41be05770d4/app/javascript/onboarding/components/EmailListTermsConditionsForm.jsx#L50) as param.

I still want to accept this and make changes to the [controller](https://github.com/thepracticaldev/dev.to/blob/master/app/controllers/users_controller.rb#L114) to snake-case params but reverting this refactor could be more straight foward. Do you have any thought on this @jacobherrington ?

jacobherrington replied:
Firstly, we probably want a feature spec to validate that onboarding works, going forward. That's something I'm happy to tackle in another PR.

Secondly, I think we should revert these changes in this PR and handle that in a new PR; I believe there are quite a few ways to address this and I'm not sure the Rails community has settled on one. Having it in a separate PR will allow us to merge this one and discuss it at length.

It's definitely a little weird to see snake case in JavaScript, but it's also weird to see camel case in Ruby.

I found [an answer to this on SO](https://stackoverflow.com/a/30557924/6010917), but I'm not sure it's the best route.

jacobherrington commented on app/javascript/onboarding/components/EmailListTermsConditionsForm.jsx:
```diff
@@ -13,10 +13,10 @@ class EmailTermsConditionsForm extends Component {
+      checkedCodeOfConduct: false,
+      checkedTermsAndConditions: false,
+      emailMembershipNewsletter: true,
+      emailDigestPeriodic: true,
```
I think we should avoid making these snake_case to camelCase changes in this PR and tackle it in another PR based on @maestromac's feedback. Would you mind undoing this change? I'd love to merge in the other linting changes.

daningenthron replied:
No problem at all, thanks for reviewing @jacobherrington! Resubmitting now.

jacobherrington commented on app/javascript/onboarding/components/SlideContent.jsx:
```diff
@@ -12,17 +17,17 @@ const SlideContent = ({ imageSource, imageAlt, content, style = { textAlign: 'ce
     <p>
       <strong>
-        <em>Let's get started...</em>
+        <em>Let`&apos;`s get started...</em>
```
We don't need the backticks here, if we leave them in, it shows up like this: 
![image](https://user-images.githubusercontent.com/11466782/66509268-4b412580-ea98-11e9-983b-bea48beebcf2.png)

jacobherrington commented on app/javascript/onboarding/components/PersonalInfoForm.jsx:
```diff
@@ -66,10 +66,10 @@ class PersonalInfoForm extends Component {
           location,
-          employer_name,
-          employment_title,
+          employerName,
```
This information doesn't seem to be making it through, this probably needs to go back to snake case as well.

daningenthron commented:
Hi @jacobherrington, removed the additional camelCase changes, and the backticks as requested. `PersonalInfoForm.jsx` has merged changes that have gone in since the original commit, with the variable `setupFormTextField` declared outside the class. ESLint was throwing a 'missing PropTypes' error on this function's params; I added a second PropTypes validator, but this appears to be the only JSX file in the repo with 2 separate validators. Would you recommend a different fix?

daningenthron commented:
Hi @jacobherrington just wanted to check in one more time - should I go ahead and close this PR at this point?

jacobherrington commented:
Hey @daningenthron it looks like there are some conflicts. If you rebase this branch, we can see what changes should be merged.

maestromac commented:
Sorry for letting this sit @daningenthron . I'm going to resolve the merge conflict and do the necessary work to get this merged.

daningenthron commented:
Thanks so much @maestromac & @jacobherrington! Wasn't sure about moving this forward bc of the time and the refactor - thanks for finishing it up!
